app/uk/gov/hmrc/tai/viewModels/TaxSummaryLabel.scala

```diff
@@ -28,29 +31,33 @@ case class HelpLink(value: String, href: String, id: String)
 
 
 object TaxSummaryLabel {
-  def apply(taxComponentType: TaxComponentType, employmentId: Option[Int], companyCarBenefits: Seq[CompanyCarBenefit], employmentIdNameMap: Map[Int, String])(implicit messages: Messages): TaxSummaryLabel = {
+  def apply(taxComponentType: TaxComponentType, employmentId: Option[Int], taxFreeAmountDetails: TaxFreeAmountDetails, amount: BigDecimal)(implicit messages: Messages): TaxSummaryLabel = {
 
     val labelString = describe(taxComponentType,
       employmentId,
-      companyCarBenefits,
-      employmentIdNameMap)
+      taxFreeAmountDetails.companyCarBenefits,
+      taxFreeAmountDetails.employmentIdNameMap)
 
-    val labelLink = createLabelLink(taxComponentType)
+    val labelLink = createLabelLink(taxComponentType, amount, taxFreeAmountDetails.totalTax)
 
     TaxSummaryLabel(labelString, labelLink)
   }
 
-  private def createLabelLink(taxComponentType: TaxComponentType)(implicit messages: Messages): Option[HelpLink] = {
+
+  private def createLabelLink(taxComponentType: TaxComponentType, amount: BigDecimal, totalTax: TotalTax)(implicit messages: Messages): Option[HelpLink] = {
+
+    val underpaymentAmount = TaxAmountDueFromUnderpayment.amountDue(amount, totalTax)
+
```

Comment:
Bug is here, the underpayAmount should only be calculated when we're either a `UnderPaymentFromPreviousYear` or a `EstimatedTaxYouOweThisYear`. Easy fix is just to turn it into a def but I guess there is a missing failing unit test to be placed over this first. To do this, we will need to create a `totalTax` object in the spec that has no `incomeCategories` which will` throw new RuntimeException("Failed to calculate the tax amount due")` and recover with a empty label in the `createLabelLink` function

Reply:
Should be done now if you can re review it